sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/SyncToken.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.appconfiguration.implementation;
+
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+
+/**
+ * SyncToken class requires id, value and sequence number
+ */
+public final class SyncToken {
+    private String id, value;
+    private long sequenceNumber;
+    private String syncTokenString;
+
+    /**
+     * Create an instance of SyncToken class
+     *
+     * @param id ID of token
+     * @param value Value of Token
+     * @param sequenceNumber Token sequence number (version). Higher value means newer version of the same token.
+     */
+    private SyncToken(String id, String value, long sequenceNumber, String syncTokenString) {
+        this.id = id;
+        this.value = value;
+        this.sequenceNumber = sequenceNumber;
+        this.syncTokenString = syncTokenString;
+    }
+
+    static SyncToken fromSyncTokenString(String syncToken) {
+        if (CoreUtils.isNullOrEmpty(syncToken)) {
+            return null;
+        }
+        try {
+            final String[] syncTokenParts = syncToken.split(";", 2);
+            // Not a fully formed Sync-Token
+            if (syncTokenParts.length != 2) {
+                return null;
```

Comment:
Log message would be nice to understand why it wasn't successfully parsed.

Reply:
updated